### PR TITLE
Add build scripts and Godeps file

### PR DESCRIPTION
**Pull request by rossmcdonald**

### CODE REVIEW

**Comment by rossmcdonald** [comment]
@joelegasse Do you mind if I go ahead and merge this in? Only changes are related to build scripts so that community members can build their own packages if needed.

**Comment by joelegasse** [comment]
I would suggest removing `support@influxdb.com` as the maintainer. This is not an officially supported product.

**Comment by rossmcdonald** [comment]
Good catch, the maintainer and vendor information has been removed.

File: build.py [hunk 1]
```diff
@@ -0,0 +1,1000 @@
...
+        shutil.copyfile(INIT_SCRIPT, os.path.join(build_root, SCRIPT_DIR[1:], INIT_SCRIPT.split('/')[1]))
+        os.chmod(os.path.join(build_root, SCRIPT_DIR[1:], INIT_SCRIPT.split('/')[1]), 0o644)
+        shutil.copyfile(SYSTEMD_SCRIPT, os.path.join(build_root, SCRIPT_DIR[1:], SYSTEMD_SCRIPT.split('/')[1]))
+        os.chmod(os.path.join(build_root, SCRIPT_DIR[1:], SYSTEMD_SCRIPT.split('/')[1]), 0o644)
+        shutil.copyfile(LOGROTATE_SCRIPT, os.path.join(build_root, LOGROTATE_DIR[1:], "influxdb-relay"))
+        os.chmod(os.path.join(build_root, LOGROTATE_DIR[1:], "influxdb-relay"), 0o644)
+        shutil.copyfile(DEFAULT_CONFIG, os.path.join(build_root, CONFIG_DIR[1:], "influxdb-relay.conf"))
+        os.chmod(os.path.join(build_root, CONFIG_DIR[1:], "influxdb-relay.conf"), 0o644)
+
+def run_generate():
```
Comment (joelegasse) [inline build.py]: The relay doesn't use `go generate` this looks like just a leftover from borrowing influxdb's build script

File: build.py [hunk 2]
```diff
@@ -0,0 +1,1000 @@
...
+    """Check user path for required dependencies.
+    """
+    logging.info("Checking for dependencies...")
+    for req in prereqs:
+        if not check_path_for(req):
+            logging.error("Could not find dependency: {}".format(req))
+            return False
+    return True
+
+def upload_packages(packages, bucket_name=None, overwrite=False):
```
Comment (joelegasse) [inline build.py]: This should probably be removed as well.

**Comment by joelegasse** [comment]
I also noticed a lot of bits still referencing `rc` and `beta` that won't apply to the relay. I'm not fond of the idea of just re-using the script as-is.

Can you clean it up so that it's stripped down for the purposes of allowing people to build their own packages?

File: build.py (hunk 1 above)
Comment (rossmcdonald) [inline build.py]: Fixed.

**Comment by rossmcdonald** [comment]
@joelegasse Sure thing. I've removed most (if not all) functionality that doesn't apply to relay, including references to RC's, betas, testing, generating, uploads, and signing. The version scheme is also to just use the git hash since relay most likely won't ever be versioned. 

Let me know if there's anything else you want me to strip out.

**Comment by joelegasse** [comment]
There are still references to signatures, minor versions, nightly vs. release... Maybe I'm overthinking this, but it just seems wrong to have all these codepaths that are never meant to be executed.

**Comment by rossmcdonald** [comment]
@joelegasse Sorry, I missed that last piece of signature functionality (now removed). 

I mean I can try stripping out everything that will never apply, but at this point the script will be able to do what it's needed for and will provide a somewhat simple way for people who are unfamiliar with Go to create packages on their own. I don't see the reason for crippling the script to the point where it only does the bare minimum. Also community members may want to use some of the extra functionality for setting up their own build processes if they so wish in the future.

**Comment by joelegasse** [comment]
I guess that's acceptable. Feel free to merge when you're ready. 👍

**Comment by rossmcdonald** [comment]
Okay, great. Thanks for your help!

**Pull request merged**